core/settings/qPrefDisplay.cpp

```diff
@@ -35,7 +34,7 @@ void qPrefDisplay::set_divelist_font(const QString& value)
 
 	if (newValue != prefs.divelist_font &&
 		!subsurface_ignore_font(qPrintable(newValue))) {
-		COPY_TXT(divelist_font, value);
+		qPrefPrivate::instance()->copy_txt(&prefs.divelist_font, value);
```

Comment:
1) Why would we need an object-instance to copy a text? This seems fundamentally wrong, unless we count the number of bytes copied or something?
2) Wouldn't it be possible to derive from a private base-class that provides these helper functions?

Reply:
1. it is fundamentally wirng, oversight from my side, corrected.
2. yes but then we would create 2 Q_OBJECTS, and have a qsettings variable in the interface classes. We would also introduce multiple inheritance something which I prefer to aviod.

Reply:
> yes but then we would create 2 Q_OBJECTS, and have a qsettings variable in the interface classes. We would also introduce multiple inheritance something which I prefer to aviod.

This explanation doesn't make sense to me:
- I don't see why this would mean multiple inheritance (which is in many cases perfectly alright): QObject > BaseClass > DerivedClass.
- The whole point of interface classes is to have some kind-of multiple inheritance in languages which don't allow it (in my opinion for dubious reasons).
- Also note that - strictly speaking - an interface is a class where all functions are pure virtual. So your class is not an interface in the classical OO sense.

Ultimately, I believe that the best course of action would be removing `copy_txt` from class context make it a helper function. This could be in a private context but even `qthelper.cpp` would be fine. As far as I can tell, this function is not only needed  in preferences code. But perhaps `set_string` would be a better name. The copying is only half of the function.

Reply:
The class would inherit from QObject and the common class.

it is not true that interface classes need to have pure virtual functions, you forget the very common interface-implement design. 

I do agree that set_string or set_text_property would be a better name, I can correct thatin a later commit. If I remove it from the class context it is no longer privat (see my use of friend). 

It is not used anywhere else, but I can easily move it to qthelper if wanted.

Reply:
> The class would inherit from QObject and the common class.

No! The base (common) class derives from `QObject` - then the derived classes are automatically `QObjects`. They don't derive again from `QObject`. That would be a disaster from Qt's object ownership point of view (which is already less-than-pretty to start with).